src/App.jsx

```diff
@@ -1,33 +1,153 @@
 import MyForm from './MyForm';
 import TodoForm from './features/TodoForm';
 import TodoList from './features/TodoList/TodoList';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 
 function App() {
   const [todolist, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
 
-  const addTodo = (title) => {
-    const newTodo = {
-      title: title,
-      id: Date.now(),
-      isCompleted: false,
+  const url = `https://api.airtable.com/v0/${import.meta.env.VITE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+  const token = `Bearer ${import.meta.env.VITE_PAT}`;
+
+  // ✅ Load todos on mount
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
+
+      const options = {
+        method: "GET",
+        headers: { Authorization: token },
+      };
+
+      try {
+        const resp = await fetch(url, options);
+        if (!resp.ok) throw new Error(resp.statusText || "Failed to fetch todos");
+
+        const { records } = await resp.json();
+
+        const fetchedTodos = records.map((record) => {
+          const todo = { id: record.id, ...record.fields };
+          if (!todo.isCompleted) todo.isCompleted = false;
+          return todo;
+        });
+
+        setTodoList(fetchedTodos);
+      } catch (error) {
+        setErrorMessage(error.message);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  // ✅ Add new todo
+  const addTodo = async (title) => {
+    const newTodo = { title, isCompleted: false };
+
+    const payload = {
+      records: [
+        {
+          fields: {
+            title: newTodo.title,
+            isCompleted: newTodo.isCompleted,
+          },
+        },
+      ],
+    };
+
+    const options = {
+      method: "POST",
+      headers: {
+        Authorization: token,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(payload),
     };
-    setTodoList([...todolist, newTodo]);
+
+    try {
+      setIsSaving(true);
+
+      const resp = await fetch(url, options);
+      if (!resp.ok) throw new Error(resp.statusText || "Failed to save todo");
+
+      const { records } = await resp.json();
+
+      const savedTodo = {
+        id: records[0].id,
+        ...records[0].fields,
+      };
+
+      if (!savedTodo.isCompleted) savedTodo.isCompleted = false;
```

Comment:
`isCompleted` should be checked against the incoming object `records`. If for some reason `isCompleted` is not spread onto`savedTodos` this would lead to some issues.